### PR TITLE
Expose non-UK Biobank subset in browser

### PR DESCRIPTION
**Pull request by phildarnowsky-broad**
Here we add the non-UK Biobank subset of v4.1 to the dataset selector in the gene, variant, region, and transcript pages.

This also includes a refactoring of the dataset metadata to write out the metadata for each dataset explicitly, rather than having a function per metadata field that takes the dataset name as input--that was proving to be unsustainable. This approach is bulkier but much more straightforward and easier to maintain.

### CODE REVIEW

**Comment by phildarnowsky-broad** [comment]
Note to self: get real sample counts before merging

**Comment by phildarnowsky-broad** [comment]
resolves #1448

File: dataset-metadata/metadata.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ export const datasetLabels = {
   gnomad_sv_r4: 'gnomAD SVs v4.0',
   gnomad_cnv_r4: 'gnomAD CNVs v4.0',
   gnomad_r4: 'gnomAD v4.0.0',
+  gnomad_r4_non_ukb: 'gnomAD 4.1 (non-UKB)',
```
Comment (rileyhgrant) [inline dataset-metadata/metadata.ts:22]: Very minor -- based on the others in the list, I think we should add a "v" and maybe make it "4.1.0" , as in "`gnomad v4.1.0 (...`" just for consistency with the other entries in the list here

Screenshot 2024-04-18 at 08 59 12

File: graphql-api/src/queries/variant-datasets/gnomad-v4-variant-queries.ts [hunk 2]
```diff
@@ -14,12 +14,14 @@ import largeGenes from '../helpers/large-genes'
 
 const GNOMAD_V4_VARIANT_INDEX = 'gnomad_v4_variants'
 
+type Subset = 'all' | 'non_ukb'
```
Comment (rileyhgrant) [inline graphql-api/src/queries/variant-datasets/gnomad-v4-variant-queries.ts:17]: Nice on continually improving the types

**Review by rileyhgrant — APPROVED** [review]
LGTM!

Looked through commit by commit and only found one tiny thing -- an inconsistency in style of dataset selector name.

Nice on continuing to improve the type system, and on this subset. All the relevant pages (gene, transcript, region, variant) looked good in my local environment to my eyes.

**Pull request merged**